File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        dict_num = {}
        for num in nums:
            if dict_num.get(num) is None:
                dict_num[num] = 1
            else:
                dict_num[num] = num + 1
                return True
        
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {   
        HashMap<Integer, Integer> hMap = new HashMap<>();

        for(int i = 0 ; i < nums.length ; i++) {
            hMap.put(nums[i], i);
        }

        for(int i = 0 ; i < nums.length ; i++) {
            int diff = target - nums[i];
            if(hMap.containsKey(diff)) {
                if(i == hMap.get(diff)) continue;
                return new int[]{i, hMap.get(diff)};
            }
        }

        return new int[]{};
    }
}
